core/src/main/java/org/mule/routing/AsynchronousUntilSuccessfulProcessingStrategy.java

```diff
@@ -361,7 +361,15 @@ private void retrieveAndProcessEvent(final Serializable eventStoreKey) throws Ob
     {
         final MuleEvent persistedEvent = getUntilSuccessfulConfiguration().getObjectStore().retrieve(eventStoreKey);
         final MuleEvent mutableEvent = threadSafeCopy(persistedEvent);
-        processEvent(mutableEvent);
+        try
+        {
+            processEvent(mutableEvent);
```

Comment:
the caller of this maethod already handles exceptions. Why not handle this tehre?

Reply:
The caller method doesn't have access to the mutable event that is created here.
I could move the event creation code to the caller and change the method signature, but I am not sure if it's worth it